ranger/core/filter_stack.py

```diff
@@ -65,6 +75,101 @@ def __str__(self):
         return "<Filter: mimetype =~ /{}/>".format(self.pattern)
 
 
+@stack_filter("hash")
+class HashFilter(BaseFilter, FileManagerAware):
+    def __init__(self, filepath):
+        self.filepath = filepath if filepath else self.fm.thisfile.path
+        if not self.filepath:
+            self.fm.notify("Error: No file selected for hashing!", bad=True)
```

Comment:
Ranger's ways are mysterious : )
But yes, in an empty directory, or one without `x` permission this can happen iirc.
I'd rather have a warning that's never hit than no warning for something that's unlikely to happen.